backend/src/main/validation/PromotionValidation.ts

```diff
@@ -20,17 +32,30 @@ export class PromotionValidation {
       .required(),
     name: Joi.string().required(),
     description: Joi.string().required(),
-    expirationDate: Joi.date().required(), // note incoming format is a string, and Joi will automatically convert to type Date
+    /**
+     * startDate is NOT required. If is null/undefined, then default to current date.
+     * However, Promotion.startDate cannot be null or undefined
+     * * note incoming format is a string, and Joi will automatically convert to type Date
+     * * startDate cannot be larger than expirationDate
+     * */
+    startDate: Joi.date()
+      .max(Joi.ref('expirationDate'))
+      .empty(null)
```

Comment:
Just curious but what does `.empty `do?

Reply:
Oh essentially, if the startDate is either undefined or null, then we default to the current date. `empty(null)` just says we consider `null` as undefined, so then Joi would default that to the current date. 

![image](https://user-images.githubusercontent.com/49849754/103178990-7ef2a180-483c-11eb-843a-b99957196220.png)

Reply:
Ooooh woah that's cool- thanks! :)